packages/react/src/components/Breadcrumb/Breadcrumb.base.tsx

```diff
@@ -47,131 +41,139 @@ const nonActionableItemProps: Partial<IContextualMenuItemProps> = {
   },
 };
 
-/**
- * {@docCategory Breadcrumb}
- */
-export class BreadcrumbBase extends React.Component<IBreadcrumbProps, any> {
-  public static defaultProps: IBreadcrumbProps = {
-    items: [],
-    maxDisplayedItems: 999,
-    overflowIndex: 0,
-  };
-
-  private _classNames: IProcessedStyleSet<IBreadcrumbStyles>;
-  private _focusZone = React.createRef<FocusZone>();
-
-  constructor(props: IBreadcrumbProps) {
-    super(props);
-
-    initializeComponentRef(this);
-    this._validateProps(props);
-  }
-
-  /**
-   * Sets focus to the first breadcrumb link.
-   */
-  public focus(): void {
-    if (this._focusZone.current) {
-      this._focusZone.current.focus();
-    }
-  }
-
-  public render(): JSX.Element {
-    this._validateProps(this.props);
-
-    const {
-      onReduceData = this._onReduceData,
-      onGrowData = this._onGrowData,
-      overflowIndex,
-      maxDisplayedItems,
-      items,
-      className,
-      theme,
-      styles,
-    } = this.props;
-    const renderedItems = [...items];
-    const renderedOverflowItems = renderedItems.splice(overflowIndex!, renderedItems.length - maxDisplayedItems!);
-    const breadcrumbData: IBreadcrumbData = {
-      props: this.props,
-      renderedItems,
-      renderedOverflowItems,
-    };
-
-    this._classNames = getClassNames(styles, {
-      className,
-      theme: theme!,
-    });
+const useComponentRef = (props: IBreadcrumbProps, focusZone: React.RefObject<FocusZone>) => {
+  React.useImperativeHandle(
+    props.componentRef,
+    () => ({
+      focus() {
+        if (focusZone.current) {
+          focusZone.current.focus();
+        }
+      },
+    }),
+    [focusZone],
+  );
+};
 
-    return (
-      <ResizeGroup
-        onRenderData={this._onRenderBreadcrumb}
-        onReduceData={onReduceData}
-        onGrowData={onGrowData}
-        data={breadcrumbData}
-      />
-    );
-  }
+export const BreadcrumbBase: React.FunctionComponent<IBreadcrumbProps> = React.forwardRef<
+  HTMLDivElement,
+  IBreadcrumbProps
+>((props, ref) => {
+  const focusZone = React.useRef<FocusZone>(null);
 
   /**
-   * Remove the first rendered item past the overlow point and put it and the end the overflow set.
+   * Remove the first rendered item past the overflow point and put it and the end the overflow set.
    */
-  private _onReduceData = (data: IBreadcrumbData): IBreadcrumbData | undefined => {
-    let { renderedItems, renderedOverflowItems } = data;
-    const { overflowIndex } = data.props;
-
-    const movedItem = renderedItems[overflowIndex!];
+  const reduceData = (data: IBreadcrumbData): IBreadcrumbData | undefined => {
+    const movedItem = renderedItems[data.props.overflowIndex!];
 
     if (!movedItem) {
       return undefined;
     }
 
-    renderedItems = [...renderedItems];
-    renderedItems.splice(overflowIndex!, 1);
+    data.renderedItems = [...data.renderedItems];
```

Comment:
You probably shouldn't be mutating the original `data` object. That's why the previous code was putting these back in at the end.